code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -244,8 +244,10 @@
 	if(!silent)
 		owner.visible_message(span_warning("[owner] catches fire!"), span_userdanger("You're set on fire!"))
 
-	if(firelight_type)
-		firelight_ref = WEAKREF(new firelight_type(owner))
+	if(moblight_type)
+		if(moblight)
+			qdel(moblight)
+		moblight = new moblight_type(owner)
```

Comment:
moblights can fail to init and selfdel if their loc is not a mob, are we guaranteed of that by this point?

Reply:
Fire status effect will not tick before an owner is assigned.